feat: Add MTA-STS support for outbound mail (#3592)
tomav/docker-mailserver: docs/content/config/advanced/mail-mta-sts.md

---
title: 'Advanced | MTA-STS'
---

> MTA-STS is an optional mechanism for a domain to signal support for
> STARTTLS. It can be used to prevent man-in-the-middle-attacks hiding the
> feature to force mail servers to send outgoing emails as plain text.
> MTA-STS is an alternative to DANE without the need of DNSSEC.
> 
> MTA-STS is supported by some of the biggest mail providers like Google Mail
> and Outlook.

Reviewer: ```suggestion
MTA-STS is an optional mechanism for a domain to signal support for STARTTLS.

- It can be used to prevent man-in-the-middle-attacks from hiding STARTTLS support that would force DMS to send outbound mail through an insecure connection.
- MTA-STS is an alternative to DANE without the need of DNSSEC.
- MTA-STS is supported by some of the biggest mail providers like Google Mail and Outlook.
```

> ## Supporting MTA-STS for outgoing mails

Reviewer: ```suggestion
## Supporting MTA-STS for outbound mail
```

> This is enabled by setting `ENABLE_MTA_STS=1` [](../environment.md#enable_mta_sts)
> in the environment.

Reviewer: ```suggestion
Enable this feature via the ENV setting [`ENABLE_MTA_STS=1`](../environment.md#enable_mta_sts).
```

> !!! warning
> 
>     MTA-STS will by default override DANE if both are in used by a domain.
>     This can be partially addressed by configuring a dane-only policy resolver
>     before the MTA-STS entry in smtp_tls_policy_maps. See [the postfix-mta-sts-resolver documentation](https://github.com/Snawoot/postfix-mta-sts-resolver#warning-mta-sts-policy-overrides-dane-tls-authentication)
>     for further details.

Reviewer: ```suggestion
!!! warning "If you have configured DANE"

    Enabling MTA-STS will by default override DANE if both are configured for a domain.

    This can be partially addressed by configuring a dane-only policy resolver before the MTA-STS entry in `smtp_tls_policy_maps`. See the [`postfix-mta-sts-resolver` documentation][postfix-mta-sts-resolver::dane] for further details.

[postfix-mta-sts-resolver::dane]: https://github.com/Snawoot/postfix-mta-sts-resolver#warning-mta-sts-policy-overrides-dane-tls-authentication
```

> ## Supporting MTA-STS for incoming mails
> 
> A good introduction can be found on [dmarcian.com](https://dmarcian.com/mta-sts/).

Reviewer: ```suggestion
## Supporting MTA-STS for inbound mail

While this feature in DMS supports ensuring STARTTLS is used when mail is sent to another mail server, you may setup similar for mail servers sending mail to DMS.

This requires configuring your DNS and hosting the MTA-STS policy file via a webserver. A good introduction can be found on [dmarcian.com](https://dmarcian.com/mta-sts/).

```

---

> title: 'Advanced | MTA-STS'

Reviewer: ```suggestion
title: 'Best Practices | MTA-STS'
hide:
  - toc # Hide Table of Contents for this page
```

Reviewer: Please also rename this file to `mta-sts.md`, and move the file to `docs/content/config/best-practices/`.

You'll need to also add an entry here for it to be added to the actual docs:

https://github.com/docker-mailserver/docker-mailserver/blob/d988d8a8d1be8bd07c271eaa9911f15f5cd033c7/docs/mkdocs.yml#L124-L126

Add this entry after the DKIM one:

```yaml
- 'MTA-STS': config/best-practices/mta-sts.md
```
